source/particles/particle_handler.cc

```diff
@@ -323,11 +347,20 @@ namespace Particles
           particle_iterator(ghost_particles, particles_in_cell.first),
           particle_iterator(ghost_particles, particles_in_cell.second));
       }
+    else if (cell->is_locally_owned())
+      {
+        const auto particles_in_cell = particles.equal_range(level_index);
+        return boost::make_iterator_range(
+          particle_iterator(particles, particles_in_cell.first),
+          particle_iterator(particles, particles_in_cell.second));
+      }
+    else
+      AssertThrow(false,
+                  ExcMessage("You can't ask for the particles on an artificial "
+                             "cell since we don't know what exists on these "
+                             "kinds of cells."));
```

Comment:
This makes the test `particles/particle_handler_06` fail,
https://cdash.43-1.org/testDetails.php?test=45322622&build=6728
@bangerth can you have a look?

Reply:
Yes, this is already #10351 .